front/src/applications/stdcm/utils/formatStdcmConf.ts

```diff
@@ -1,22 +1,39 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import type { TFunction } from 'i18next';
 import { compact } from 'lodash';
 import type { Dispatch } from 'redux';
 
 import type {
+  PathItemLocation,
   PathfindingItem,
   PostTimetableByIdStdcmApiArg,
   TrainScheduleBase,
 } from 'common/api/osrdEditoastApi';
 import type { InfraState } from 'reducers/infra';
 import { setFailure } from 'reducers/main';
-import type { OsrdStdcmConfState, StandardAllowance } from 'reducers/osrdconf/types';
+import type { OsrdStdcmConfState, StandardAllowance, StdcmPathStep } from 'reducers/osrdconf/types';
 import { dateTimeFormatting } from 'utils/date';
 import { kmhToMs, mToMm, tToKg } from 'utils/physics';
 import { ISO8601Duration2sec, sec2ms } from 'utils/timeManipulation';
 
 import createMargin from './createMargin';
 
+const getStepLocation = (step: StdcmPathStep): PathItemLocation => {
```

Comment:
This would also be useful for operational studies if this took a `PathStep` instead, but I suppose the longer term goal is to keep these two completely separate?

Reply:
Yes :/ but maybe I can do smth like this:
```suggestion
const getStepLocation = (step: PathStep | StdcmPathStep): PathItemLocation => {
```

Reply:
Ah yes, that sounds like a workable solution!

For reference, the duplicated code: https://github.com/OpenRailAssociation/osrd/blob/86ad6408b1b1a567b691882dc1c3d543b4df9112/front/src/modules/pathfinding/utils.ts#L68